tests/simple_example.rs

```diff
@@ -105,7 +105,7 @@ where
 }
 
 async fn init_server<'a, T>(
-    remote: &'a mut EphemeralRepository<Json>,
+    remote: &'a mut Repository<EphemeralRepository, Json>,
```

Comment:
What do you think about just passing in `EphemeralRepository` and not wrap it in `Repository`? The only thing that buys us is that we call `Repository::check` to see if the metadata path matches the role, but I'm not sure if that check actually buys us that much. If a server messed up and wrote the root metadata as `timestamp.json`, the client would fetch that file then it should fail to deserialize into `TimestampMetadata`. So really, this check is more for making sure we don't generate a malformed repository, but given how low-level the current implementation is, the whole thing probably needs to be architected (see #232).

Reply:
Not quite.  The reason I used Repository in these test binaries was because its method signatures were nicer to use (I could store RawMetadata instead of having to convert them to bytes first).